// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: file/service/v1/file.proto

package servicev1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on OssUploadUrlRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *OssUploadUrlRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on OssUploadUrlRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// OssUploadUrlRequestMultiError, or nil if none found.
func (m *OssUploadUrlRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *OssUploadUrlRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Method

	if m.ContentType != nil {
		// no validation rules for ContentType
	}

	if m.BucketName != nil {
		// no validation rules for BucketName
	}

	if m.FilePath != nil {
		// no validation rules for FilePath
	}

	if m.FileName != nil {
		// no validation rules for FileName
	}

	if len(errors) > 0 {
		return OssUploadUrlRequestMultiError(errors)
	}

	return nil
}

// OssUploadUrlRequestMultiError is an error wrapping multiple validation
// errors returned by OssUploadUrlRequest.ValidateAll() if the designated
// constraints aren't met.
type OssUploadUrlRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m OssUploadUrlRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m OssUploadUrlRequestMultiError) AllErrors() []error { return m }

// OssUploadUrlRequestValidationError is the validation error returned by
// OssUploadUrlRequest.Validate if the designated constraints aren't met.
type OssUploadUrlRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OssUploadUrlRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OssUploadUrlRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OssUploadUrlRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OssUploadUrlRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OssUploadUrlRequestValidationError) ErrorName() string {
	return "OssUploadUrlRequestValidationError"
}

// Error satisfies the builtin error interface
func (e OssUploadUrlRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOssUploadUrlRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OssUploadUrlRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OssUploadUrlRequestValidationError{}

// Validate checks the field values on OssUploadUrlResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *OssUploadUrlResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on OssUploadUrlResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// OssUploadUrlResponseMultiError, or nil if none found.
func (m *OssUploadUrlResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *OssUploadUrlResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for UploadUrl

	// no validation rules for DownloadUrl

	// no validation rules for ObjectName

	// no validation rules for FormData

	if m.BucketName != nil {
		// no validation rules for BucketName
	}

	if len(errors) > 0 {
		return OssUploadUrlResponseMultiError(errors)
	}

	return nil
}

// OssUploadUrlResponseMultiError is an error wrapping multiple validation
// errors returned by OssUploadUrlResponse.ValidateAll() if the designated
// constraints aren't met.
type OssUploadUrlResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m OssUploadUrlResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m OssUploadUrlResponseMultiError) AllErrors() []error { return m }

// OssUploadUrlResponseValidationError is the validation error returned by
// OssUploadUrlResponse.Validate if the designated constraints aren't met.
type OssUploadUrlResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OssUploadUrlResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OssUploadUrlResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OssUploadUrlResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OssUploadUrlResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OssUploadUrlResponseValidationError) ErrorName() string {
	return "OssUploadUrlResponseValidationError"
}

// Error satisfies the builtin error interface
func (e OssUploadUrlResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOssUploadUrlResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OssUploadUrlResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OssUploadUrlResponseValidationError{}

// Validate checks the field values on GetDownloadUrlRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetDownloadUrlRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetDownloadUrlRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetDownloadUrlRequestMultiError, or nil if none found.
func (m *GetDownloadUrlRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *GetDownloadUrlRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return GetDownloadUrlRequestMultiError(errors)
	}

	return nil
}

// GetDownloadUrlRequestMultiError is an error wrapping multiple validation
// errors returned by GetDownloadUrlRequest.ValidateAll() if the designated
// constraints aren't met.
type GetDownloadUrlRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetDownloadUrlRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetDownloadUrlRequestMultiError) AllErrors() []error { return m }

// GetDownloadUrlRequestValidationError is the validation error returned by
// GetDownloadUrlRequest.Validate if the designated constraints aren't met.
type GetDownloadUrlRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetDownloadUrlRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetDownloadUrlRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetDownloadUrlRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetDownloadUrlRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetDownloadUrlRequestValidationError) ErrorName() string {
	return "GetDownloadUrlRequestValidationError"
}

// Error satisfies the builtin error interface
func (e GetDownloadUrlRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetDownloadUrlRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetDownloadUrlRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetDownloadUrlRequestValidationError{}

// Validate checks the field values on GetDownloadUrlResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetDownloadUrlResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetDownloadUrlResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetDownloadUrlResponseMultiError, or nil if none found.
func (m *GetDownloadUrlResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *GetDownloadUrlResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return GetDownloadUrlResponseMultiError(errors)
	}

	return nil
}

// GetDownloadUrlResponseMultiError is an error wrapping multiple validation
// errors returned by GetDownloadUrlResponse.ValidateAll() if the designated
// constraints aren't met.
type GetDownloadUrlResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetDownloadUrlResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetDownloadUrlResponseMultiError) AllErrors() []error { return m }

// GetDownloadUrlResponseValidationError is the validation error returned by
// GetDownloadUrlResponse.Validate if the designated constraints aren't met.
type GetDownloadUrlResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetDownloadUrlResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetDownloadUrlResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetDownloadUrlResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetDownloadUrlResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetDownloadUrlResponseValidationError) ErrorName() string {
	return "GetDownloadUrlResponseValidationError"
}

// Error satisfies the builtin error interface
func (e GetDownloadUrlResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetDownloadUrlResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetDownloadUrlResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetDownloadUrlResponseValidationError{}

// Validate checks the field values on ListFileRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *ListFileRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ListFileRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ListFileRequestMultiError, or nil if none found.
func (m *ListFileRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *ListFileRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if m.Bucket != nil {
		// no validation rules for Bucket
	}

	if m.Folder != nil {
		// no validation rules for Folder
	}

	if m.Recursive != nil {
		// no validation rules for Recursive
	}

	if len(errors) > 0 {
		return ListFileRequestMultiError(errors)
	}

	return nil
}

// ListFileRequestMultiError is an error wrapping multiple validation errors
// returned by ListFileRequest.ValidateAll() if the designated constraints
// aren't met.
type ListFileRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ListFileRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ListFileRequestMultiError) AllErrors() []error { return m }

// ListFileRequestValidationError is the validation error returned by
// ListFileRequest.Validate if the designated constraints aren't met.
type ListFileRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListFileRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListFileRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListFileRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListFileRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListFileRequestValidationError) ErrorName() string { return "ListFileRequestValidationError" }

// Error satisfies the builtin error interface
func (e ListFileRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListFileRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListFileRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListFileRequestValidationError{}

// Validate checks the field values on ListFileResponse with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *ListFileResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ListFileResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ListFileResponseMultiError, or nil if none found.
func (m *ListFileResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *ListFileResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return ListFileResponseMultiError(errors)
	}

	return nil
}

// ListFileResponseMultiError is an error wrapping multiple validation errors
// returned by ListFileResponse.ValidateAll() if the designated constraints
// aren't met.
type ListFileResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ListFileResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ListFileResponseMultiError) AllErrors() []error { return m }

// ListFileResponseValidationError is the validation error returned by
// ListFileResponse.Validate if the designated constraints aren't met.
type ListFileResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListFileResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListFileResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListFileResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListFileResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListFileResponseValidationError) ErrorName() string { return "ListFileResponseValidationError" }

// Error satisfies the builtin error interface
func (e ListFileResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListFileResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListFileResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListFileResponseValidationError{}

// Validate checks the field values on DeleteFileRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *DeleteFileRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DeleteFileRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// DeleteFileRequestMultiError, or nil if none found.
func (m *DeleteFileRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *DeleteFileRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if m.Bucket != nil {
		// no validation rules for Bucket
	}

	if m.Object != nil {
		// no validation rules for Object
	}

	if len(errors) > 0 {
		return DeleteFileRequestMultiError(errors)
	}

	return nil
}

// DeleteFileRequestMultiError is an error wrapping multiple validation errors
// returned by DeleteFileRequest.ValidateAll() if the designated constraints
// aren't met.
type DeleteFileRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DeleteFileRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DeleteFileRequestMultiError) AllErrors() []error { return m }

// DeleteFileRequestValidationError is the validation error returned by
// DeleteFileRequest.Validate if the designated constraints aren't met.
type DeleteFileRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DeleteFileRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DeleteFileRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DeleteFileRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DeleteFileRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DeleteFileRequestValidationError) ErrorName() string {
	return "DeleteFileRequestValidationError"
}

// Error satisfies the builtin error interface
func (e DeleteFileRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDeleteFileRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DeleteFileRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DeleteFileRequestValidationError{}

// Validate checks the field values on DeleteFileResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *DeleteFileResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DeleteFileResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// DeleteFileResponseMultiError, or nil if none found.
func (m *DeleteFileResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *DeleteFileResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return DeleteFileResponseMultiError(errors)
	}

	return nil
}

// DeleteFileResponseMultiError is an error wrapping multiple validation errors
// returned by DeleteFileResponse.ValidateAll() if the designated constraints
// aren't met.
type DeleteFileResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DeleteFileResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DeleteFileResponseMultiError) AllErrors() []error { return m }

// DeleteFileResponseValidationError is the validation error returned by
// DeleteFileResponse.Validate if the designated constraints aren't met.
type DeleteFileResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DeleteFileResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DeleteFileResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DeleteFileResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DeleteFileResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DeleteFileResponseValidationError) ErrorName() string {
	return "DeleteFileResponseValidationError"
}

// Error satisfies the builtin error interface
func (e DeleteFileResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDeleteFileResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DeleteFileResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DeleteFileResponseValidationError{}
